addons/block_code/simple_nodes/simple_character/simple_character.gd

```diff
@@ -31,6 +31,27 @@ const PLAYER_KEYS = {
 	}
 }
 
+const PLAYER_JOYSTICK_BUTTONS = {
+	"player_1": {
+		0: {
+			"up": JOY_BUTTON_DPAD_UP,
+			"down": JOY_BUTTON_DPAD_DOWN,
+			"left": JOY_BUTTON_DPAD_LEFT,
+			"right": JOY_BUTTON_DPAD_RIGHT,
+		},
+	},
+	
+	"player_2":
+	{
+		1: {
+			"up": JOY_BUTTON_DPAD_UP,
+			"down": JOY_BUTTON_DPAD_DOWN,
+			"left": JOY_BUTTON_DPAD_LEFT,
+			"right": JOY_BUTTON_DPAD_RIGHT,
+		}
+	}
+}
```

Comment:
I understand that you've done this for consistency with the keyboard actions above but I don't think you need two copies of the actions: I think you could do:

```
const JOYPAD_BUTTONS = {
    "up": JOY_BUTTON_DPAD_UP,
   ...
}
```

and iterate it once per player.

Reply:
right, that would be much cleaner. fixed in c502b4a